agent/pkg/cruntimes/crutils.go

```diff
@@ -27,6 +28,17 @@ import (
 
 var logLevel = regexp.MustCompile("(?P<level>INFO|WARN|ERROR|FATAL):    (?P<log>.*)")
 
+// BaseTempDirName returns a per-user directory name that is unique
+// for the user, but consistently named between agent runs.
+func BaseTempDirName() string {
+	user, err := user.Current()
+	if err != nil {
+		panic(err)
+	}
+
+	return fmt.Sprintf("/tmp/determined-%s", user.Username)
```

Comment:
what if the same user has many agents? does it make more sense to use `opts.AgentID` instead?

Reply:
I can add that as well to the mix as well, but buy default that is set to hostname, so it doesn't prevent multiple users from conflicting over time -- so to avoid file name/permissions conflicts between multiple users over time on a compute node, it minimally needs to be unique for each user.   

Another option is to just drop the folder altogether.  It appears you are using it to cleanup left overs from prior runs on startup, but that isn't as useful in an HPC environment where there are lots of nodes and you may or may not have been running on this particularly node previously.   It may help avoid a full /tmp if prior agent shutdown did not cleanup well.

Reply:
> Another option is to just drop the folder altogether. It appears you are using it to cleanup left overs from prior runs on startup, but that isn't as useful in an HPC environment where there are lots of nodes and you may or may not have been running on this particularly node previously. It may help avoid a full /tmp if prior agent shutdown did not cleanup well.

i'll trust your judgement on that. i think it is removed on graceful shutdown right now, but idk if we handle that cases well in general now, anyway

Reply:
>  i think it is removed on graceful shutdown right now, but idk if we handle that cases well in general now, anyway

Yes, and that is definitely the most valuable part.  It doesn't hurt to try to attempt to do so otherwise, but it does mean we need a per-user, per-agentdID path.    Since we have that covered now, should be OK.